Sources/SwiftSyntax/gyb_generated/syntax_nodes/SyntaxNodes.swift

```diff
@@ -6901,6 +6901,947 @@ extension ObjCSelectorPieceSyntax: CustomReflectable {
+    }
+    // Check child #2 child is FunctionDeclNameSyntax 
+    assert(rawChildren[2].raw != nil)
+    if let raw = rawChildren[2].raw {
+      let info = rawChildren[2].syntaxInfo
+      let absoluteRaw = AbsoluteRawSyntax(raw: raw, info: info)
+      let syntaxData = SyntaxData(absoluteRaw, parent: Syntax(self))
+      let syntaxChild = Syntax(syntaxData)
+      assert(syntaxChild.is(FunctionDeclNameSyntax.self))
+    }
+    // Check child #3 child is TokenSyntax or missing
+    if let raw = rawChildren[3].raw {
+      let info = rawChildren[3].syntaxInfo
+      let absoluteRaw = AbsoluteRawSyntax(raw: raw, info: info)
+      let syntaxData = SyntaxData(absoluteRaw, parent: Syntax(self))
+      let syntaxChild = Syntax(syntaxData)
+      assert(syntaxChild.is(TokenSyntax.self))
+    }
+  }
+}
+
+extension DifferentiableAttributeFuncSpecifierSyntax: CustomReflectable {
+  public var customMirror: Mirror {
+    return Mirror(self, children: [
+      "label": Syntax(label).as(SyntaxProtocol.self),
+      "colon": Syntax(colon).as(SyntaxProtocol.self),
+      "functionDeclName": Syntax(functionDeclName).as(SyntaxProtocol.self),
+      "trailingComma": trailingComma.map(Syntax.init)?.as(SyntaxProtocol.self) as Any,
+    ])
+  }
+}
+
+// MARK: - FunctionDeclNameSyntax
+
+/// A function declaration name (e.g. `foo(_:_:)`).
+public struct FunctionDeclNameSyntax: SyntaxProtocol, SyntaxHashable {
+  enum Cursor: Int {
+    case name
+    case arguments
+  }
+
+  public let _syntaxNode: Syntax
+
+  /// Converts the given `Syntax` node to a `FunctionDeclNameSyntax` if possible. Returns
+  /// `nil` if the conversion is not possible.
+  public init?(_ syntax: Syntax) {
+    guard syntax.raw.kind == .functionDeclName else { return nil }
+    self._syntaxNode = syntax
+  }
+
+  /// Creates a `FunctionDeclNameSyntax` node from the given `SyntaxData`. This assumes
+  /// that the `SyntaxData` is of the correct kind. If it is not, the behaviour
+  /// is undefined.
+  internal init(_ data: SyntaxData) {
+    assert(data.raw.kind == .functionDeclName)
+    self._syntaxNode = Syntax(data)
+  }
+
+  /// 
+  /// The base name of the referenced function.
+  /// 
+  public var name: Syntax {
+    get {
+      let childData = data.child(at: Cursor.name,
+                                 parent: Syntax(self))
+      return Syntax(childData!)
+    }
+    set(value) {
+      self = withName(value)
+    }
+  }
+
+  /// Returns a copy of the receiver with its `name` replaced.
+  /// - param newChild: The new `name` to replace the node's
+  ///                   current `name`, if present.
+  public func withName(
+    _ newChild: Syntax?) -> FunctionDeclNameSyntax {
+    let raw = newChild?.raw ?? RawSyntax.missing(SyntaxKind.unknown)
+    let newData = data.replacingChild(raw, at: Cursor.name)
+    return FunctionDeclNameSyntax(newData)
+  }
+
+  /// 
+  /// The argument labels of the referenced function, optionally                    specified.
```

Comment:
Note: this description comment is suboptimal and will be fixed in a follow-up.
Tracked here: https://github.com/apple/swift/pull/27506#discussion_r345309203